swap_meet/vendor.py

```diff
@@ -1,2 +1,87 @@
+from swap_meet.item import Item
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item:Item) -> Item:
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item:Item) -> list or bool:
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category:str)->list:
+        list_of_items_with_specific_category = []
+        for each_item in self.inventory:
+            if each_item.category == category:
+                list_of_items_with_specific_category.append(each_item)
+        return list_of_items_with_specific_category
+    
+    def swap_items(self, other, my_item: Item, their_item: Item)->bool:
```

Comment:
I see that you used type hints here. I would be curious to know what made you choose to use type hints. I would also suggest adding a comment here on what is and why you chose to use it. Just so I can understand your thought process when reading through your code.